lib/web/users.go

```diff
@@ -220,6 +225,51 @@ func getUsers(ctx context.Context, m userAPIGetter) ([]ui.UserListEntry, error)
 	return uiUsers, nil
 }
 
+// listUsersResponse is the response for the list users request.
+type listUsersResponse struct {
+	// Items is the list of users retrieved.
+	Items []ui.UserListEntry `json:"items"`
+	// StartKey is the position from which to resume search.
+	StartKey string `json:"startKey"`
+}
+
+// listUsers returns a paginated list of users.
+func listUsers(ctx context.Context, clt userAPIGetter, values url.Values) (*listUsersResponse, error) {
+	limit, err := QueryLimitAsInt32(values, "limit", defaults.MaxIterationLimit)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	resp, err := clt.ListUsers(ctx, &userspb.ListUsersRequest{
+		PageSize:  limit,
+		PageToken: values.Get("startKey"),
+		Filter: &types.UserFilter{
+			SearchKeywords: client.ParseSearchKeywords(values.Get("search"), ' '),
+		},
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var uiUsers []ui.UserListEntry
+	for _, u := range resp.GetUsers() {
+		// Do not display system users in the WebUI
+		if types.IsSystemResource(u) {
+			continue
+		}
```

Comment:
This will alter the page size if a system user is returned on a given page, right? Perhaps we could implement this filtering on the API level instead?